analysis/quantifier-comparisons/10-miQC-filtering.Rmd

```diff
@@ -0,0 +1,234 @@
+
+### Testing Posterior Probability Cutoffs
+
+miQC will remove any cells that have a high probability of being compromised, by default these are cells with higher than 0.75 posterior probability. 
+We can increase that threshold, which means that more cells will be kept and see if we are able to recover some of the cells that were being thrown out previously that we think should have been included (ie. cells with low mito that are being thrown out). 
+
+```{r}
+# change the posterior cutoff for miQC to keep throw away cells with > 0.9 probability of being compromised
+purrr::iwalk(cr_like_sce, model_sce, posterior_cutoff = 0.9)
+```
+It doesn't appear that increasing the probability threshold rescues very many more cells with low mito content in SCPCR000126 and SCPCR000127 that are being filtered out. 
+
+### Testing different mixture models 
+
+In addition to altering the threshold used for filtering, we can also test the use of different mixture models to calculate the posterior probability of a cell being compromised. 
+We will test use of both the spline and polynomial modes rather than the default linear model. 
+The polynomial model uses a single polynomial to model the entire datset, while spline uses a piecewise continuous function of many polynomials to model the data set. 
+
+```{r}
+# test spline mixture model rather than linear 
+purrr::iwalk(cr_like_sce, model_sce, model_type = "spline")
+```
+
+Use of the spline model definitely doesn't seem ideal with one sample, SCPCR000126, showing the reverse of what we would expect and only keeping cells with low mito content. 
+For the other samples we see a similar trend that we had observed in the linear model. 
+
+```{r}
+# test polynomial mixture model rather than linear
+purrr::iwalk(cr_like_sce, model_sce, model_type = "polynomial")
+```
+
+For the linear model, we still see that cells with low mito are getting excluded from SCPCR000118 and SCPCR000119, but now see that same trend in SCPCR000220 and SCPCR000221 as well. 
+It appears to model the cells with very high number of unique genes and low mito content as cells to exclude when we would rather keep those. 
+
+## Filter Cells
+
+Now we can compare filtering using miQC vs. using the pre-determined cutoffs to see how the different filtering effects the overall number of cells that are removed. 
+We can also look at the distribution of each of the metrics used for filtering, genes detected/cell, and mito content/cell to see how filtering affects the population of cells. 
+Here we are using a threshold of 500 genes detected per cell and 10% mitochondrial reads.
+
+```{r}
+# function to filter sce using miQC 
+miQC_filter <- function(sce, model_type = "linear", posterior_cutoff = 0.75){
+  sce_model <- miQC::mixtureModel(sce, model_type)
+  filtered_sce <- miQC::filterCells(sce, sce_model)
+}
+
+# filter using manual 
+manual_filter <- function(sce, 
+                          mito_cutoff = 10, 
```

Comment:
Just curious why you chose to bump this down to 10%? I thoguth the 20% cutoff made sense as a default (for single cell data at least).

Reply:
Do we want to include a few different cutoffs (maybe w/ and without the `genes_detected_cutoff`) so we can see where miQC falls within those comparisons?

Reply:
From a previous comment you made about moving the cutoff down to 10%, I had chosen to make that change (and based on our conversation in the scpca meeting that we wanted to keep everything below 10% mito). I think it would be fairly reasonable to add in a few different cutoffs to see where miQC falls, but I'm not entirely sure if that will change the decision to use miQC or not and think maybe that question falls under the next steps question of "how do we decide what additional filtering if any to include for the `ccdl_suggests` column"?